apiproduct-operations/README.md

```diff
@@ -0,0 +1,202 @@
+| creator | `POST /contracts`                                                            |
+| admin   | `GET /contracts`, `GET /contracts/*`, `POST /contracts`, `POST /contracts/*` |
+
+
+## Credentials are "the key" for checking the Operation
+
+Configuring the set of API Products with the various operations is one part of
+the set up.  The next step is to grant access to these products to different
+applications - this is done via app credentials. In Apigee, you need to
+configure an App authorized for the product, to get a set of credentials.  This
+is typically done in the self-service developer portal (aka API Catalog). After
+obtaining credentials, the developer embeds the credentials into the app, and
+sends those credentials along with API requests.
+
+When the API request reaches Apigee, the API proxy that handles it must invoke a
+policy to verify the credentials - either
+[VerifyAPIKey](https://cloud.google.com/apigee/docs/api-platform/reference/policies/verify-api-key-policy)
+or
+[OAuthV2/VerifyAccessToken](https://cloud.google.com/apigee/docs/api-platform/reference/policies/oauthv2-policy#verifyaccesstoken). When
+that happens, Apigee checks the credential and maps it to one or more API
+Products. (In the simple case, an app is authorized for a single API Product,
+but Apigee allows apps to have access to more than one product.)  Then, Apigee
+checks that the current executing operation - using the REST model, this is a
+verb+path combination - is included within one of the API Products that is
+authorized for that credential. If not, then the request is rejected.
+
+In our example, for a `GET /contracts` request, the credential must be authorized
+for the viewer or admin product. For a `POST /contracts` request, the credential
+must be authorized for either creator or admin.
+
+## Implementation in the API Proxy
+
+This sample uses a simple API Proxy to demonstrate this function. The proxy uses
+a
+[VerifyAPIKey](https://cloud.google.com/apigee/docs/api-platform/reference/policies/verify-api-key-policy)
+policy to identify the calling application and the API product associated with
+the proxy. All of the magic of authorizing the operation for the given
+credential is done _implicitly_ by the VerifyAPIKey policy. There's nothing else
+you need to do in the API Proxy to make that happen.
+
+For the purposes of demonstration, the proxy uses an
+[AssignMessage](https://cloud.google.com/apigee/docs/api-platform/reference/policies/assign-message-policy)
+policy to set a mock success response, which includes the API product name, and
+the operation that was authorized.
+
+There's also a conditional flow in the proxy that uses [OAuthV2/VerifyAccessToken](https://cloud.google.com/apigee/docs/api-platform/reference/policies/oauthv2-policy#verifyaccesstoken) in place of VerifyAPIKey.
+The behavior with regard to operations authorization checks is exactly the same.
+
+
+## Screencast
```

Comment:
I would recommend removing this section and adding it when you have the video ready

Reply:
fixed with latest commit